### DISCUSSION TITLE
Duplicate node

### DISCUSSION THREAD

**Discussion by Lumengine** [skypjack/entt#1046] [General]
Hi,
I'm using EnTT with NodeComponent 
```
	struct NodeComponent
	{
		Uint32 childrenCount{};
		IEntity first{entt::null};
		IEntity prev{entt::null};
		IEntity next{entt::null};
		IEntity parent{entt::null};
	};
```
and I want to duplicate a node with all the children. 
Of course, children can have a lot of different components and the duplicated node should be a sibling of the duplicating node.
Any tips or advice of how to do that properly ?

Best,

Baptiste

**Comment by skypjack** [comment]
Do your entities all have the same components? Otherwise they need to be treated individually and in that case you can look at the example in the repo to copy a single entity, then apply it to your hierarchy. If, on the other hand, they all have the same components, it's easier because you can collect the entities and copy all the elements one storage at a time probably.

**Reply by Lumengine** [reply]
Unfortunately, they don't. So, I will look at the example ! Thank you !

**Reply by skypjack** [reply]
Well, not a big deal anyway. You can create a new entity and copy components as needed in the different storage.
You don't even have to know the actual C++ types. Just make sure that your types are copyable, otherwise you're going to have a problem. 🙂
